File: algorithms/Palindromic Substrings/Palindromic Substrings.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int count = s.size();
        for(int i = 0; i < s.size(); i++) {
            int len1 = expand(s, i, i+1);
            int len2 = expand(s, i, i);
            
			// every palindrome found has len/2 total number of palindromes
            count += len1/2;
            count += len2/2;
        }
        return count;
    }
    
    int expand(string s, int i, int j) {
        while(i >= 0 && j < s.size() && s[i] == s[j]) {
            i--;j++;
        }
        return j-i-1;
    }
};

File: algorithms/P/Palindromic Substrings/Palindromic Substrings.java
int n = s.length();
        int count = 0;
        for(int i = 0;i<n;i++){
            int l = i;
            int r = l;
            while(l>=0 && r<n && s[l]==s[r]){
                count++;
                l--;
                r++;
            }
        }
        for(int i = 0;i<n-1;i++){
            int l =  i;
            int r = l+1;
            while(l>=0 && r<n && s[l]==s[r]){
                count++;
                l--;
                r++;
            }
        }
        return count;